frontend/utils/useApi.tsx

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+import json from "../api/endpoint.json"
+import {baseURL} from "../api/url.json"
+const useApi = () => {
+  const [data, setData] = useState<object>({}); //saves data as json
+  const [endpoints,setEndpoints] = useState<endpoint>(json)
+  function get(end: string) : void
+  {
+    const url =  baseURL+checkForEndpoint(end,"GET")
+    console.log(url)
+    axios.get(url).then(function (response: any) {
+      setData(response.data)
+    }).catch(function (error) {
+      throw new Error(error);
+    })
+  }
+
+  function post(end: string) : void 
+  {
+    const url = baseURL+checkForEndpoint(end,"POST")
+    axios.post(url).then(function (response: any) {
+      setData(response.data)      
+    }).catch(function (error) {
+      throw new Error(error);
+    })
+  }
+
+  function checkForEndpoint(end:string,method:string): string
+  {
+    
+    if(!(end in endpoints)) throw new Error(`No endpoint found! Avileble: ${JSON.stringify(endpoints)}`);
```

Comment:
A wrong endpoint should also return JSON, for example, `{"error": "content of the error"}`, but this is something to handle on the backend. Then, we can consider improving the error handling on the front.